File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let answer = 0;
    let [i, j] = [0, people.length-1];
    
    people.sort((a, b) => a - b);
    
    while (i <= j) {
        if (people[i] + people[j] <= limit) i++;
        answer++;
        j--;
    }
    
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    i, j = 0, len(people)-1
    
    people.sort()
    
    while i <= j:
        if people[i] + people[j] <= limit:
            i += 1
        answer += 1
        j -= 1
    
    return answer
